### PR TITLE
Use out parameter for generated methods returning JSVal

### LINKED ISSUE #6307: Use a MutableHandleValue outparam for calls returning GCThings
This is a followup to https://github.com/servo/servo/pull/6150 .

DOM methods that return JSVal or JSObjects are better served by taking a MutableHandleValue to dump the return value into.

### PR DESCRIPTION
**Pull request by jdm**
Returning raw JSVal values makes it easier to create GC hazards in code that calls these methods. Accepting a MutableHandle argument instead ensures that the values are rooted by the caller. As a side benefit, this also allows us to remove a bunch of awkward macro use in the WebGL code. Depends on https://github.com/servo/mozjs/pull/519.

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #6307
- [x] There are tests for these changes

### CODE REVIEW

File: components/script/dom/dissimilaroriginwindow.rs [hunk 1]
```diff
@@ -167,9 +167,9 @@ impl DissimilarOriginWindowMethods for DissimilarOriginWindow {
     }
 
     // https://html.spec.whatwg.org/multipage/#dom-opener
-    fn Opener(&self, _: JSContext) -> JSVal {
+    fn Opener(&self, _: JSContext, mut retval: MutableHandleValue) {
         // TODO: Implement x-origin opener
-        UndefinedValue()
+        retval.set(UndefinedValue());
```
Comment (gterzian) [inline components/script/dom/dissimilaroriginwindow.rs:172]: is `retval` not undefined by default?

**Review by gterzian — APPROVED** [review]
LGTM

File: components/script/dom/history.rs [hunk 2]
```diff
@@ -289,11 +289,12 @@ impl History {
 
 impl HistoryMethods for History {
     /// <https://html.spec.whatwg.org/multipage/#dom-history-state>
-    fn GetState(&self, _cx: JSContext) -> Fallible<JSVal> {
+    fn GetState(&self, _cx: JSContext, mut retval: MutableHandleValue) -> Fallible<()> {
```
Comment (sagudev) [inline components/script/dom/history.rs:292]: I understand why this needs to be mut for https://doc.servo.org/mozjs/gc/root/struct.MutableHandle.html#tymethod.set, but this is still weird as MutableHandle is clonable so `&mut` does not help a lot with safety.

File: components/script/dom/dissimilaroriginwindow.rs (hunk 1 above)
Comment (sagudev) [reply components/script/dom/dissimilaroriginwindow.rs:172]: Not necessarily, that depends on the caller (although in this case the codegen). I think this is more correct and being more explicit is always better.

File: components/script/dom/history.rs (hunk 2 above)
Comment (sagudev) [reply components/script/dom/history.rs:292]: I opened https://github.com/servo/mozjs/issues/520

**Comment by sagudev** [comment]
BTW to caught clippy stuff locally you can use updated configuration for IDE: https://book.servo.org/hacking/editor-support.html#visual-studio-code

**Pull request merged**